client/websocket_client.tsx

```diff
@@ -157,7 +170,7 @@ export default class WebSocketClient {
         }
     }
 
-    sendMessage(action, data, responseCallback) {
+    sendMessage(action: string, data: any, responseCallback?: () => void) { // TODO data any or unknown?
```

Comment:
I think `any` here is fine since we know it can really be any type we want. `unknown` I think is more used for when you're checking for the type and acting accordingly.